web/src/components/snapshots/SnapshotStorageDestination.tsx

```diff
@@ -77,36 +77,107 @@ type State = {
   tmpFileSystemNFSServer?: string;
   tmpFileSystemType?: string;
   updatingSettings?: boolean;
-  useIamAws?: boolean;
+  useIamAws: boolean;
 };
 
 type StoreProviderName = "aws" | "gcp" | "azure" | "other";
-type StoreProvider = {
-  [K in StoreProviderName]?: {
-    accessKeyID?: string;
```

Comment:
Rather than having one type that represents all of the storage provider types, break up into separate storage provider types. This should prevent things like assigning a invalid keys to specific providers like `region` to `gcp`